### PR TITLE
Remove the 'app' domain from flutter daemon

### LINKED ISSUE #6658: remove launch functionality from flutter daemon
Once  [IJ to uses `flutter run --machine` rather than `flutter daemon`](https://github.com/flutter/flutter-intellij/issues/401) to launch and interact with Flutter applications, we can remove launch functionality from `flutter daemon`.

### PR DESCRIPTION
**Pull request by DanTup**
By default the daemon won't register the "app" domain, you need to opt-in (which the 'run' command does, as well as the tests for the app functionality).

Fixes #6658.

This removes all of the `app` functionality from `daemon`, not just started, since without being able to start the app none of the others made sense.

We may want to review the overlap between `daemon`/`run --machine` going forwards. It seems like there's a lot less of it now (one is certainly not a subset of the other) so it might make sense that they're separate daemon classes (with a shared base).

### CODE REVIEW

File: packages/flutter_tools/doc/daemon.md [hunk 1]
```diff
@@ -10,7 +10,7 @@ flutter daemon
 
 It runs a persistent, JSON-RPC based server to communicate with devices. IDEs and other tools can start the flutter tool in this mode and get device addition and removal notifications, as well as being able to programmatically start and stop apps on those devices.
 
-A subset of the `flutter daemon` commands/events are also exposed via `flutter run --machine` which allows IDEs and tools to launch flutter applications and interact to send commands like Hot Reload. Which commands/events are available in this mode is documented at the bottom of this document.
+A set of `flutter daemon` commands/events are also exposed via `flutter run --machine` which allows IDEs and tools to launch flutter applications and interact to send commands like Hot Reload. Which commands/events are available in this mode is documented at the bottom of this document.
```
Comment (devoncarew) [inline packages/flutter_tools/doc/daemon.md]: 'Which commands/events are available in this mode is documented at the bottom of this document.'

===>

`The command and events that are available in this mode are documented at the bottom of this document.`

File: packages/flutter_tools/doc/daemon.md [hunk 2]
```diff
@@ -76,23 +76,7 @@ It is up to the client to decide how best to display the message; for some clien
...
-- `mode`: One of either `debug`, `profile`, or `release`.
-- `target`: Optional; the target file to start.
-- `hot`: Optional; whether to start the application using `--hot` mode
-
-On success, returns a map with the fields:
-- `appId`: this is is used when sending app events, and can be used by clients to stop the app (`app.stop`).
-- `deviceId`
-- `directory`
-- `supportsRestart`
+The app domain is only available via `flutter run --machine` and not in `flutter daemon`.
```
Comment (devoncarew) [inline packages/flutter_tools/doc/daemon.md]: I would strike this, and only remove the `start` (`and `discover`) commands.

File: packages/flutter_tools/lib/src/commands/daemon.dart [hunk 3]
```diff
@@ -83,11 +83,14 @@ class Daemon {
...
     this.daemonCommand,
     this.notifyingLogger,
-    this.logToStdout = false
+    this.logToStdout = false,
+    this.enableAppDomain = false,
   }) {
     // Set up domains.
     _registerDomain(daemonDomain = new DaemonDomain(this));
-    _registerDomain(appDomain = new AppDomain(this));
+    if (enableAppDomain) {
```
Comment (devoncarew) [inline packages/flutter_tools/lib/src/commands/daemon.dart]: I'd remove this conditional.

File: packages/flutter_tools/lib/src/commands/daemon.dart [hunk 4]
```diff
@@ -537,22 +488,6 @@ class AppDomain extends Domain {
...
   }
 
-  Future<List<Map<String, dynamic>>> discover(Map<String, dynamic> args) async {
-    final String deviceId = _getStringArg(args, 'deviceId', required: true);
-
-    final Device device = await daemon.deviceDomain._getDevice(deviceId);
-    if (device == null)
-      throw "device '$deviceId' not found";
-
-    final List<DiscoveredApp> apps = await device.discoverApps();
```
Comment (devoncarew) [inline packages/flutter_tools/lib/src/commands/daemon.dart]: I'm curious if `device.discoverApps()` is called from anywhere now? If no, we should add a todo: to the declaration.

We'll need similar but different functionality for add2app.

File: packages/flutter_tools/doc/daemon.md (hunk 2 above)
Comment (DanTup) [reply packages/flutter_tools/doc/daemon.md]: Ah, I assumed we couldn't do `restart`/`discover` if the app wasn't launched within this process. If that's not the case then this was a bit overkill, I'll put them back!

**Review by devoncarew — APPROVED** [review]

File: packages/flutter_tools/lib/src/commands/daemon.dart (hunk 4 above)
Comment (DanTup) [reply packages/flutter_tools/lib/src/commands/daemon.dart]: Ok, I've added a TODO and a link here. I'll land this with that TODO and come back to it depending on the response from @jason-simmons.

**Pull request merged**